### PR TITLE
UX: chat message thread indicator v2 design & refactor

### PR DESCRIPTION
**Pull request by chapoi**
This PR contains:
* gridified the thread message indicator, alleviating some problems with positioning and overflow
* participant avatars will overlap/smush on smaller size and mobile
* the excerpt went from 3 > 2 lines of wrapping on smaller size, still 1 line on large size
* dropped the copy of "last reply"
* fixed wrong line height
* moved the "x replies" over to the right near the participants, as that makes more sense
* using a bubble to indicate other participants, instead of copy

### desktop & drawer
image

image

image

### mobile
identical to drawer/small screen

And before anyone asks: it IS centered, so misalignment you may see is an optical illusion.
image

And as a footnote: this PR introduces the `@container` query, which is experimental. Nothing will break when it's being viewed in a not-supported browser, but it will be less elegant.

### CODE REVIEW

File: plugins/chat/assets/javascripts/discourse/components/chat-message-thread-indicator.hbs [hunk 1]
```diff
@@ -8,50 +8,26 @@
...
-    <Chat::Thread::Participants @thread={{@message.thread}} />
+  <div class="chat-message-thread-indicator__last-reply-info">
+    <span class="chat-message-thread-indicator__last-reply-username">
+      {{@message.thread.preview.lastReplyUser.username}}
+    </span>
+    <span class="chat-message-thread-indicator__last-reply-timestamp">
+      {{format-date @message.thread.preview.lastReplyCreatedAt leaveAgo="true"}}
+    </span>
+  </div>
+  {{! <div class="chat-message-thread-indicator__participants"> }}
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/chat-message-thread-indicator.hbs:27]: did you mean to remove this?

**Comment by jjaffeux** [comment]
On specific width of drawer I can end up in this case:

Screenshot 2023-06-21 at 16 48 13

**Comment by jjaffeux** [comment]
Number and avatar don't have same size:

Screenshot 2023-06-21 at 16 51 12
Screenshot 2023-06-21 at 16 51 07

**Comment by jjaffeux** [comment]
Screenshot 2023-06-21 at 16 51 58

Shouldn't we try to align left of the thread indicator with the reaction/message above?

**Comment by jjaffeux** [comment]
(on mobile at least) Alignment feels off to me here:

Screenshot 2023-06-21 at 16 54 23


Also I wonder if dont want just a little bit more spacing between text of the message and above content.

**Comment by jjaffeux** [comment]
@jordanvidrine could you review this please? I was making feedback, but I think it would be better if another designer was doing it.

File: plugins/chat/assets/javascripts/discourse/components/chat-message-thread-indicator.hbs (hunk 1 above)
Comment (chapoi) [reply plugins/chat/assets/javascripts/discourse/components/chat-message-thread-indicator.hbs:27]: Remove what?

**Comment by chapoi** [comment]
> Screenshot 2023-06-21 at 16 49 17
> 
> I guess it's related to your is-online question. Avatar is not left aligned here.

It is

**Comment by chapoi** [comment]
> Screenshot 2023-06-21 at 16 49 17
> 
> I guess it's related to your is-online question. Avatar is not left aligned here.

It is though. But visually, because of the transparent border it looks 1px misaligned. If we decide to keep is-online, I'll go add in a negative margin.

**Comment by chapoi** [comment]
> Screenshot 2023-06-21 at 16 51 58
> 
> Shouldn't we try to align left of the thread indicator with the reaction/message above?

No? Because then you don't have any padding on the side…?

**Comment by chapoi** [comment]
> (on mobile at least) Alignment feels off to me here, I know it's not supposed to be related or on the same line, but the fact that's it "almost" on the same line makes it look off IMO:
> 
> Screenshot 2023-06-21 at 16 54 23
> 
> Also I wonder if we dont want just a little bit more spacing between text of the message and above content.

Yeah I considered adding in more space, but to keep it optimised I didn't. Because it's only the avatar that's close, if I add more space, the text I felt it looked wrong too between username and text.

I disagree with the alignment being a problem. Like you said, it's a different part of the element, it shouldn't align.

File: plugins/chat/assets/javascripts/discourse/components/chat-message-thread-indicator.hbs (hunk 1 above)
Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/components/chat-message-thread-indicator.hbs:27]: The comment? `{{! <div class="chat-message-thread-indicator__participants"> }}` ?

**Comment by jjaffeux** [comment]
> No? Because then you don't have any padding on the side…?

I guess I would agree if left avtar side was aligned with above message:

Screenshot 2023-06-21 at 18 28 03

Something like that?

**Comment by chapoi** [comment]
> On specific width of drawer I can end up in this case:
> 
> Screenshot 2023-06-21 at 16 48 13

fixed


> (on mobile at least) Alignment feels off to me here, I know it's not supposed to be related or on the same line, but the fact that's it "almost" on the same line makes it look off IMO:
> 
> Screenshot 2023-06-21 at 16 54 23

added space




> Screenshot 2023-06-21 at 18 28 03
> 
> Something like that? Anyways, just feedback, do what you think is best.

This Im deciding against. Aligning the block is more important, that's also how we do it for oneboxes and blockquotes.

**Comment by jjaffeux** [comment]
> This Im deciding against. Aligning the block is more important, that's also how we do it for oneboxes and blockquotes.

Bookmarked to complain abut this again in 6 months 🔖

**Comment by jordanvidrine** [comment]
I'll be taking a look at these today during travel.

Here are the first two things I see from the OP. Just want to make sure @chapoi that you updated the OP images?

The #1 here is not aligned with letters
image

What is the reason the furthest right avatar is larger than the rest?
image

I will have more when I get some more time to look.

**Comment by chapoi** [comment]
> I'll be taking a look at these today during travel.
> 
> Here are the first two things I see from the OP. Just want to make sure @chapoi that you updated the OP images?
> 
> The #1 here is not aligned with letters image
> 
> What is the reason the furthest right avatar is larger than the rest? image
> 
> I will have more when I get some more time to look.

Yup, those mistakes had been pointed out and should be fixed. Ill update the images

**Review by jjaffeux — APPROVED** [review]

**Comment by jjaffeux** [comment]
@jordanvidrine I merged it because we don't want to sit on these improvements for too long, but feel free to make more comments we will do follow-ups

**Pull request merged**